From 1a606d6eba762fb8b4edc603aecc55d6761403f5 Mon Sep 17 00:00:00 2001
From: Mikko Ylinen <mikko.ylinen@intel.com>
Date: Fri, 24 May 2019 14:35:25 +0300
Subject: [PATCH 2/4] qatzip_stream: have unambiguous QZ_ERROR prints

Signed-off-by: Mikko Ylinen <mikko.ylinen@intel.com>
---
 src/qatzip_stream.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/qatzip_stream.c b/src/qatzip_stream.c
index 282879e..7f4776a 100644
--- a/src/qatzip_stream.c
+++ b/src/qatzip_stream.c
@@ -67,7 +67,7 @@ static int initStream(QzSession_T *sess, QzStream_T *strm)
     strm->opaque = malloc(sizeof(QzStreamBuf_T));
     stream_buf = (QzStreamBuf_T *) strm->opaque;
     if (NULL == stream_buf) {
-        QZ_ERROR("Fail to allocate memory for QzStreamBuf");
+        QZ_ERROR("Fail to allocate memory for QzStreamBuf (struct)");
         return QZ_FAIL;
     }
 
@@ -78,7 +78,7 @@ static int initStream(QzSession_T *sess, QzStream_T *strm)
 
     if (NULL == stream_buf->in_buf ||
         NULL == stream_buf->out_buf) {
-        QZ_ERROR("Fail to allocate memory for QzStreamBuf");
+        QZ_ERROR("Fail to allocate memory for QzStreamBuf (buffers)");
         return QZ_FAIL;
     }
     QZ_DEBUG("Allocate stream buf %u\n", stream_buf->buf_len);
-- 
2.20.1

