From af8441c6ffa37540bf507276689f2ddcde93ed5d Mon Sep 17 00:00:00 2001
From: Mikko Ylinen <mikko.ylinen@intel.com>
Date: Fri, 24 May 2019 14:35:54 +0300
Subject: [PATCH 3/4] qatzip: fix resource counting

Signed-off-by: Mikko Ylinen <mikko.ylinen@intel.com>
---
 src/qatzip.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/qatzip.c b/src/qatzip.c
index 967ae03..c7b9f2c 100755
--- a/src/qatzip.c
+++ b/src/qatzip.c
@@ -346,6 +346,7 @@ static void stopQat(void)
     (void)icp_sal_userStop();
     g_process.num_instances = (Cpa16U)0;
     g_process.qz_init_status = QZ_NONE;
+    g_process.pcie_count = -1;
 }
 
 static void freeQzMemEntries(void)
-- 
2.20.1

