From 9e7e7de68f682ae498c3024d30df971c609b37c5 Mon Sep 17 00:00:00 2001
From: Mikko Ylinen <mikko.ylinen@intel.com>
Date: Fri, 24 May 2019 14:47:53 +0300
Subject: [PATCH 4/4] qatzip_stream: qzMalloc PINNED_MEM only when HW is
 available

Signed-off-by: Mikko Ylinen <mikko.ylinen@intel.com>
---
 src/qatzip_stream.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/qatzip_stream.c b/src/qatzip_stream.c
index 7f4776a..d83b0c4 100644
--- a/src/qatzip_stream.c
+++ b/src/qatzip_stream.c
@@ -73,8 +73,8 @@ static int initStream(QzSession_T *sess, QzStream_T *strm)
 
     stream_buf->out_offset = 0;
     stream_buf->buf_len = qz_sess->sess_params.strm_buff_sz;
-    stream_buf->in_buf = qzMalloc(stream_buf->buf_len, NODE_0, PINNED_MEM);
-    stream_buf->out_buf = qzMalloc(stream_buf->buf_len, NODE_0, PINNED_MEM);
+    stream_buf->in_buf = qzMalloc(stream_buf->buf_len, NODE_0, (sess->hw_session_stat == QZ_NO_HW) ? COMMON_MEM : PINNED_MEM);
+    stream_buf->out_buf = qzMalloc(stream_buf->buf_len, NODE_0, (sess->hw_session_stat == QZ_NO_HW) ? COMMON_MEM : PINNED_MEM);
 
     if (NULL == stream_buf->in_buf ||
         NULL == stream_buf->out_buf) {
-- 
2.20.1

